provider/openstack/client.go

```diff
@@ -149,8 +209,13 @@ func (c *ClientFactory) getClientState(options ...client.Option) (client.Authent
 }
 
 // getClientByAuthMode creates a new client for the given AuthMode.
-func (c *ClientFactory) getClientByAuthMode(authMode identity.AuthMode, cred identity.Credentials, options ...client.Option) (client.AuthenticatingClient, error) {
-	newClient, err := c.clientFunc(cred, authMode, c.sslHostnameConfig.SSLHostnameVerification(), c.spec.CACertificates, options...)
+func (c *ClientFactory) getClientByAuthMode(authMode identity.AuthMode, cred identity.Credentials, options ...ClientOption) (client.AuthenticatingClient, error) {
+	newClient, err := c.clientFunc(cred, authMode,
+		append(options,
+			WithSkipHostnameVerification(!c.sslHostnameConfig.SSLHostnameVerification()),
+			WithCACertificates(c.spec.CACertificates...),
+		)...,
```

Comment:
Given that the original slice will be quite small, it is quite possible that you are mutating the caller's value here. I would suggest creating a copy as follows:

```go
append([]ClientOption{
                        WithSkipHostnameVerification(!c.sslHostnameConfig.SSLHostnameVerification()),
                        WithCACertificates(c.spec.CACertificates...),
                }, options...)...,
```